fedimint-api/src/db/mod.rs

```diff
@@ -169,6 +169,93 @@ impl Drop for CommitTracker {
     }
 }
 
+pub struct ModuleDatabaseTransaction<'a, 'b: 'a> {
+    tx: &'a mut DatabaseTransaction<'b>,
+    module_prefix: Vec<u8>,
+}
+
+impl<'a, 'b> ModuleDatabaseTransaction<'a, 'b> {
+    pub fn new(
+        dbtx: &'a mut DatabaseTransaction<'b>,
+        module_prefix: u16,
+    ) -> ModuleDatabaseTransaction<'a, 'b> {
+        ModuleDatabaseTransaction {
+            tx: dbtx,
+            module_prefix: vec![(module_prefix >> 8) as u8, (module_prefix & 0xff) as u8],
```

Comment:
Can we just use the `consensus_encode` impl for `module_prefix` here?

Reply:
Done